src/test/java/net/datafaker/providers/base/PhoneNumberTest.java

```diff
@@ -57,103 +62,139 @@ void testPhone_CA() {
                 if (!util.isValidNumberForRegion(proto, phoneNumberRegion)) {
+                    if (log.isLoggable(FINE)) {
+                        log.fine("Invalid phone: %s".formatted(phoneNumber));
+                    }
                     errorCount++;
                 }
-            } catch (Exception e) {
+            } catch (NumberParseException e) {
+                if (log.isLoggable(FINE)) {
+                    log.fine("Invalid phone: %s (caused by: %s)".formatted(phoneNumber, e));
+                }
                 errorCount++;
             }
         }
-        assertThat(errorCount).isLessThanOrEqualTo(15);
+        assertThat(errorCount).isLessThanOrEqualTo(allowedErrorsCount);
     }
 
     @ParameterizedTest
     @MethodSource("generateLanguageAndRegionOfLocales")
-    void testAllPhoneNumberInternational(Locale locale, String phoneNumberRegion) throws NumberParseException {
-        final PhoneNumberUtil util = PhoneNumberUtil.getInstance();
+    void testAllPhoneNumberInternational(Locale locale, String phoneNumberRegion, int allowedErrorsCount) throws NumberParseException {
         int errorCount = 0;
         final BaseFaker faker = new BaseFaker(locale);
         final PhoneNumber phoneNumberProvider = faker.phoneNumber();
-        for (int i = 0; i < 10; i++) {
+        for (int i = 0; i < COUNT; i++) {
             String phoneNumber = phoneNumberProvider.phoneNumberInternational();
             Phonenumber.PhoneNumber proto = util.parse(phoneNumber, phoneNumberRegion);
             if (!util.isValidNumberForRegion(proto, phoneNumberRegion)) {
+                if (log.isLoggable(FINE)) {
+                    log.fine("Invalid phone: %s".formatted(phoneNumber));
+                }
                 errorCount++;
             }
         }
-        assertThat(errorCount).isLessThan(25);
+        assertThat(errorCount).isLessThanOrEqualTo(allowedErrorsCount);
+    }
+
+    @ParameterizedTest
+    @MethodSource("generateLanguageAndRegionOfLocales")
+    void testAllPhoneNumberMobile(Locale locale, String phoneNumberRegion, int allowedErrorsCount) throws NumberParseException {
+        int errorCount = 0;
+        final BaseFaker faker = new BaseFaker(locale);
+        final PhoneNumber phoneNumberProvider = faker.phoneNumber();
+        for (int i = 0; i < COUNT; i++) {
+            String phoneNumber = phoneNumberProvider.cellPhone();
+            Phonenumber.PhoneNumber proto = util.parse(phoneNumber, phoneNumberRegion);
+            if (!util.isValidNumberForRegion(proto, phoneNumberRegion)) {
+                if (log.isLoggable(FINE)) {
+                    log.fine("Invalid phone: %s".formatted(phoneNumber));
+                }
+                errorCount++;
+            }
+        }
+        assertThat(errorCount).isLessThanOrEqualTo(allowedErrorsCount);
+    }
+
+    private static Arguments args(Locale locale, String phoneNumberRegion) {
+        return args(locale, phoneNumberRegion, (int) (COUNT * 0.77));
```

Comment:
Thought you said 66?

Reply:
may be also extract into a var?

Reply:
> Thought you said 66?

Yes, initially I put 66%, but later had to increase this number to 77% because of flaky tests failed on CI. :)

Reply:
> extract into a var

Yes, I've extract 77% to a constant.